edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent/Interfaces/IMetricsPublisher.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Microsoft.Azure.Devices.Edge.Agent.DiagnosticsComponent
```

Comment:
As a general comment, the directory structure should match the namespace. This file should live in `edge-agent/src/Microsoft.Azure.Device.Edge.Agent.Diagnostics/IMetricsPublsher.cs`. We don't typically have a separate directory for interfaces.

Subfolders in this structure are also generally all lowercase. So, if we did have happen to have a subfolder `Interfaces` it should be `interfaces`